test/test_oauth2.py

```diff
@@ -280,6 +280,44 @@ def test_it_updates_a_profile_when_exchanging(test_client: FlaskClient) -> None:
     assert str(original_profile.name) == 'Josiah Carberry'
 
 
+def test_it_rejects_a_private_name_when_exchanging(test_client: FlaskClient) -> None:
+    with requests_mock.Mocker() as mocker:
+        mocker.post('http://www.example.com/server/token',
+                    json={'access_token': '1/fFAGRNJru1FTz70BzhT3Zg', 'expires_in': 3920,
+                          'foo': 'bar', 'token_type': 'Bearer', 'orcid': '0000-0002-1825-0097',
+                          'name': ''})
+
+        response = test_client.post('/oauth2/token',
+                                    data={'client_id': 'client_id',
+                                          'client_secret': 'client_secret',
+                                          'redirect_uri': 'http://www.example.com/client/redirect',
+                                          'grant_type': 'authorization_code', 'code': '1234'})
+
+    assert response.status_code == 400
+    assert response.headers.get('Content-Type') == 'application/json'
+
+
+def test_it_ignores_now_private_names_when_exchanging(test_client: FlaskClient) -> None:
+    original_profile = Profile('a1b2c3d4', Name('Foo Bar'), '0000-0002-1825-0097')
```

Comment:
good setup that doesn't go through the original exchange but creates a local `Profile` only